pkg/capture/capturetypes/packet.go

```diff
@@ -41,23 +41,18 @@ const (
 
 )
 
-// GPPacket stores all relevant packet details for a flow
-type GPPacket struct {
-	// packet size
-	NumBytes uint32
-
-	// packet inbound or outbound on interface
-	DirInbound bool
-
-	// flag to easily determine if a packet is IPv4 or IPv6
-	IsIPv4 bool
-
-	// packet descriptors / hashes
-	EPHash        EPHash
-	EPHashReverse EPHash
-
-	// Auxilliary data (protocol dependent) for further analysis
-	AuxInfo byte
+// EPHash is a typedef that allows us to replace the type of hash
+type EPHash [37]byte
+
+// Reverse calculates the reverse of an EPHash (i.e. source / destination switched)
+func (h EPHash) Reverse() (rev EPHash) {
+	copy(rev[0:16], h[16:32])
+	copy(rev[16:32], h[0:16])
+	copy(rev[32:34], h[34:36])
+	copy(rev[34:36], h[32:34])
+	rev[36] = h[36]
+
+	return
 }
 
 // ClassifyPacketDirection is responsible for running a variety of heuristics on the packet
```

Comment:
These are now all more explicit (i.e. instead of being fed a `GPPacket` they simply take whatever parameters they require to make their decision).